numpy/core/src/multiarray/buffer.c

```diff
@@ -363,7 +363,10 @@ _buffer_format_string(PyArray_Descr *descr, _tmp_string_t *str,
         switch (descr->type_num) {
         case NPY_BOOL:         if (_append_char(str, '?')) return -1; break;
         case NPY_BYTE:         if (_append_char(str, 'b')) return -1; break;
-        case NPY_UBYTE:        if (_append_char(str, 'B')) return -1; break;
+        case NPY_UBYTE:
+        case NPY_DATETIME:
+        case NPY_TIMEDELTA:    if (_append_char(str, 'B')) return -1; break;
```

Comment:
Would it be legal to set the format to `8B` instead, and leave the shape as `()`?

Reply:
Why not 'Q' then, since we call these `datetime64` and `timedelta64`, the 64 bit is explicit. This would mean that `d=np.ones((3), 'M8[D]'); np.array(memoryview(d))` would return a `uint64` array, rather that a 3 by 8 'uint8' one.